src/shapes/ActiveSelection.ts

```diff
@@ -134,6 +135,25 @@ export class ActiveSelection extends Group {
     return false;
   }
 
+  _applyLayoutStrategy(context: LayoutContext): void {
+    super._applyLayoutStrategy(context);
+    if (this._objects.length === 0) {
+      // in this case layout was skipped
+      // we reset transform for the next selection
+      Object.assign(this, {
+        left: 0,
+        top: 0,
+        angle: 0,
+        scaleX: 1,
+        scaleY: 1,
+        skewX: 0,
+        skewY: 0,
+        flipX: false,
+        flipY: false,
+      });
+    }
+  }
```

Comment:
To keep the ref so we can listen to events etc.

Reply:
Also if somrone subclasses they can set the ref and it will remain as is

Reply:
I think that this is an evidence that keeping the same reference for ActiveSelection is a bad breaking change and design. It causes bugs and workarounds for things that were more natural behaviour. 

> To keep the ref so we can listen to events etc.

What's an use case for listening events on the ActiveSelection? Is there an alternative that doesn't require keeping the same reference?

> Also if somrone subclasses they can set the ref and it will remain as is

Not clear to me, can you make an example of use case + code?

Reply:
1. **Bug Scope**: I have ben thinking of this specific fix. Could it be that it belongs to Group? I need to test this. Might be this is a bug in `enterGroup` when group is rotated.
2. **Events**: There are many requests from devs needing to listen to active selection events. Many speak of customizing the stack order in accordance to some business logic (which can now be accomplished using `multiSelectAdd`). Also toggling selectablility of objects, responding visually to stuff etc.
3. **Subclassing**: In our project I will subclass active selection to override `multiSelectAdd` to block objects that should not be selected from being added to the selection (formerly `_createActiveSelection`, `_updateActiveSelection`)

- The only place that was creating a new `ActiveSelection` was `_createActiveSelection`.
- I want group to be robost, it seems to me this bug is a group bug.
- I also think canvas should accept the active selection ref in options or create a default one if non was passed. We can make the constructor of `Activeselection` protected - that will help devs see the change.
- Devs can also add logic to `setActiveObject` if they must continue creating active selections:
```
setActiveObject(object) {
  if (object instaceof ActiveSelection) {
    this._activeSelection = object;
    object.set('canvas', this);
  }
  super.setActiveObject(object);
}
```

I do think it makes sense holding the ref but I am not fixed on it.

Reply:
Okay I see your points now and all the use cases make sense. Just not sure about the implementation requirement of keeping the same reference.

> We can make the constructor of Activeselection protected - that will help devs see the change.

That could block use cases we don't know about. What about making `_activeObject` read-only? I'm thinking something similar to `window.focused` which is a read-only property in the DOM. What if instead of `canvas._activeSelection.on("change", fn)` we have a `SelectionManager` that will manage to subscribe to the actual ActiveSelection which can change reference? E.g.

```
canvas.selection.on("change", fn)
canvas.selection.multiSelectAdd = ...
```

So that for instance stuff like `canvas.handleMultiSelection` can be moved to `SelectionManager` as well.

Reply:
What is the diif between `SelectionManager` active selection?
I also thought about retargeting all active selection events to the canvas... that will solve the ref issue for events but not for subclassing

Reply:
> What is the diff between SelectionManager active selection?

That you can do most of the stuff you showed on the SelectionManager, which makes sense to always keep its reference the same vs on the ActiveSelection, which intuitively should be allowed to change reference.

Just as gut feeling, it also bothers me that the `_activeSelection` doesn't change reference. It could for instance cause issues with React which assumes the reference is changed if the content is changed. For instance, I would need to check `activeSelection._objects` and loop through the children to know if a selection changed its content and this could break cases where I have the selection in the Redux state for instance. (In Collaboard we have `selectedObject` in the state for instance, although not saved as direct reference to the fabric.Object indeed because of the reference issue with objects by reference)

Reply:
Legit point of view
I am agnostic
It is a breaking change, and the worst kind because it is silent
However we want to publish with as less changes as possible
@asturur I know you didn't like the const ref either. What do you think?

Reply:
i forgot to update myself on this. i ll do and let's clear this out soon

Reply:
First question, is there any reason why here we do this manual change while in _discardActiveObject we call the util `resetObjectTransform`?
We could extend resetObjectTransform to handle top and left since seems clear to me that in both cases we use resetObjectTransform we don't care for survival of top/left.

My general idea is that is late to remove this reference activeSelection even if i don't like it at all, it would be nice to reuse resetObjectTransform to save code duplication.

Reply:
As long as fabric does not create instances of active selection I am ok with dropping the ref.
A bit late, but fine. Lets take advantage of beta.
Second, the resetting should occur in group in terms of expectation, not only active selection.
I do not remeber why I didn't use resetObjectTransform but I do remember considering it.

Reply:
The only reason why i don't want to remove the ref is because i don't like go back and forth on choices.
We made this, let's move forward with this. If some issue will arise we will have to reconsider and we will do a breaking change.
i will try to reuse resetObject trasform in a separate PR, i can't see any reason for it to be a problem.